dbaas/backup/tests/test_purge_unused_exports.py

```diff
@@ -0,0 +1,159 @@
+            self.environment, export.nfsaas_path_host
+        )
+
+    def test_cannot_delete_inactive_with_active_snapshot(self):
+        export = NFSaaSHostAttr()
+        export.host = self.instance.hostname
+        export.is_active = False
+        export.save()
+
+        snapshot = SnapshotFactory(instance=self.instance)
+        snapshot.export_path = export.nfsaas_path
+        snapshot.save()
+
+        snapshot = SnapshotFactory(instance=self.instance)
+        snapshot.export_path = export.nfsaas_path
+        snapshot.purge_at = datetime.now()
+        snapshot.save()
+
+        exports = HostAttr.objects.all()
+        self.assertEqual(1, len(exports))
+        self.assertIn(export, exports)
+
+        self.assertTrue(purge_unused_exports())
+
+        exports = HostAttr.objects.all()
+        self.assertEqual(1, len(exports))
+        self.assertIn(export, exports)
+
+    @patch('backup.tasks.delete_export')
+    def test_can_delete_inactive_with_inactive_snapshot(
+        self, delete_export_method
+    ):
+        export = NFSaaSHostAttr()
+        export.host = self.instance.hostname
+        export.is_active = False
+        export.save()
+
+        snapshot = SnapshotFactory(instance=self.instance)
+        snapshot.export_path = export.nfsaas_path
+        snapshot.purge_at = datetime.now()
+        snapshot.save()
+
+        exports = HostAttr.objects.all()
+        self.assertEqual(1, len(exports))
+        self.assertIn(export, exports)
+
+        self.assertTrue(purge_unused_exports())
+
+        exports = HostAttr.objects.all()
+        self.assertEqual(0, len(exports))
+        self.assertNotIn(export, exports)
+
+        delete_export_method.assert_called()
+        delete_export_method.assert_called_once_with(
+            self.environment, export.nfsaas_path_host
+        )
+
+    @patch('backup.tasks.delete_export')
+    def test_task_with_success(self, delete_export_method):
+        export = NFSaaSHostAttr()
+        export.host = self.instance.hostname
+        export.is_active = False
+        export.save()
+
+        exports = HostAttr.objects.all()
+        self.assertEqual(1, len(exports))
+        self.assertIn(export, exports)
+
+        task = TaskHistoryFactory()
+        self.assertIsNone(task.details)
+
+        self.assertTrue(purge_unused_exports(task))
+
+        exports = HostAttr.objects.all()
+        self.assertEqual(0, len(exports))
+        self.assertNotIn(export, exports)
+
+        self.assertIn('Removing: {}'.format(export), task.details)
+        self.assertIn('Success', task.details)
+
+        delete_export_method.assert_called()
+        delete_export_method.assert_called_once_with(
+            self.environment, export.nfsaas_path_host
+        )
+
+    @patch('backup.tasks.delete_export')
+    def test_task_with_error(self, delete_export_method):
+        delete_export_method.side_effect = Exception('Fake error')
+
+        export = NFSaaSHostAttr()
```

Comment:
This code cannot be in setUp, because other tests are using this method, and for these other tests we cannot raise an exception